semgrep-core/src/osemgrep/cli_scan/Core_runner.ml

```diff
@@ -184,27 +191,47 @@ let call_semgrep_core conf all_rules all_targets =
        else config
      in
   *)
+  (* TODO: we should not need to use Common.map below, because
+   * Run_semgrep.semgrep_with_raw_results_and_exn_handler can accept
+   * a list of targets with different languages! We just
+   * need to pass the right target object (and not a lang_job)
```

Comment:
The issue I ran into was that `Run_semgrep.targets_of_config` requires the xlang object to contain a single language:

https://github.com/returntocorp/semgrep/blob/6385ca49963991f05da5d4c995fa538bb56c6af3/semgrep-core/src/runner/Run_semgrep.ml#L518

I find the xlang type confusing. Why is it sometimes used to represent a single language and other times multiple languages? I would rather have a clear distinction between those cases:
* a single programming language: `Lang.t`
* a single language that determines how a pair (pattern, target) is going to be processed: `type xlang = Lang of Lang.t | Regexp | Generic`
* multiple languages acceptable for parsing a pattern given in a rule: `{ primary: xlang; secondary: xlang list }` or even `xlang list` with runtime checks that the list is not empty.

Reply:
yep, maybe we should do that.